WalletWasabi.Fluent/ViewModels/Wallets/WalletViewModel.cs

```diff
@@ -97,19 +98,22 @@ public WalletViewModel(UiContext uiContext, IWalletModel walletModel, Wallet wal
 			{
 				Settings.SelectedTab = 0;
 				Navigate(NavigationTarget.DialogScreen).To(Settings);
-			});
+			},
+			Observable.Return(!WalletModel.IsWatchOnlyWallet));
 
 		WalletCoinsCommand = ReactiveCommand.Create(() => Navigate(NavigationTarget.DialogScreen).To().WalletCoins(WalletModel));
 
-		CoinJoinSettingsCommand = ReactiveCommand.Create(
+		var coinJoinSettingsCommand = ReactiveCommand.Create(
 			() =>
 			{
 				Settings.SelectedTab = 1;
 				Navigate(NavigationTarget.DialogScreen).To(Settings);
 			},
 			Observable.Return(!WalletModel.IsWatchOnlyWallet));
+		
+		CoinJoinSettingsCommand = coinJoinSettingsCommand;
 
-		CoinJoinStateViewModel = new CoinJoinStateViewModel(uiContext, WalletModel);
+		CoinJoinStateViewModel = new CoinJoinStateViewModel(uiContext, WalletModel, coinJoinSettingsCommand);
```

Comment:
I'm not particularly proud of this line, but I felt I need to reuse the original command to avoid having a dependency the Settings VM to set the correct SelectedTab number.

@ichthus1604 can you think of a better way?

Reply:
@SuperJMN Looks like you can remove the command completely from `WalletViewModel`, and keep it in `CoinJoinStateViewModel`.

It's not being used in XAML other than the MusicBox, and the only place where `WalletViewModel` references it is [line 240](https://github.com/SuperJMN/WalletWasabi/blob/improvements/musicbox-cj-settings/WalletWasabi.Fluent/ViewModels/Wallets/WalletViewModel.cs#L240), which could be replaced by `CoinJoinStateViewModel.CoinJoinSettingsCommand`.

Reply:
OK, I had to add a dependency on WalletSettingsViewModel to do that. Please check :)